Framework/Core/include/Framework/DataAllocator.h

```diff
@@ -91,12 +94,82 @@ class DataAllocator
   void
   adopt(const OutputSpec &spec, TObject*obj);
 
-  /// Serialize a snapshot of a TObject when called, which will then
+  /// Serialize a snapshot of a TObject when called, which will then be
   /// sent once the computation ends.
   /// Framework does not take ownership of the @a object. Changes to @a object
   /// after the call will not be sent.
-  void serializeSnapshot(const OutputSpec &spec, TObject* const object);
+  template <typename T>
+  typename std::enable_if<std::is_base_of<TObject, T>::value == true, void>::type
+  snapshot(const OutputSpec &spec, T *const object) {
```

Comment:
thinking more about this, I would make the argument a reference also here

Reply:
Given the underlying `ROOT` API works in terms of pointers, I think it's better we keep it consistent. If we do not want to, then the change will affect also TMessageSerializer, so I'd rather have it in a separate PR.